packages/kit/src/runtime/server/data/index.js

```diff
@@ -153,10 +170,113 @@ function json_response(json, status = 200) {
  * @param {Redirect} redirect
  */
 export function redirect_json_response(redirect) {
-	return json_response(
-		JSON.stringify({
-			type: 'redirect',
-			location: redirect.location
-		})
-	);
+	return json_response({
+		type: 'redirect',
+		location: redirect.location
+	});
+}
+
+export const get_data_json = to_generator(_get_data_json);
+
+/**
+ * The first chunk returns the devalue'd nodes with potentially pending promises.
+ * Subsequent chunks (if any) return the resolved promises.
+ * @param {import('types').RequestEvent} event
+ * @param {import('types').SSROptions} options
+ * @param {Array<import('types').ServerDataSkippedNode | import('types').ServerDataNode | import('types').ServerErrorNode | null | undefined>} nodes
+ * @param {(result: {has_more: boolean; data: any}, done: boolean) => void} next
+ */
+async function _get_data_json(event, options, nodes, next) {
+	let promise_id = 1;
+	let count = 0;
+	let strings = [];
+
+	try {
+		for (const node of nodes) {
+			let uses_str = '';
+
+			const revivers = {
+				/** @param {any} thing */
+				Promise: (thing) => {
+					if (typeof thing?.then === 'function') {
+						const id = promise_id++;
+						count += 1;
+
+						thing
+							.then(/** @param {any} d */ (d) => ({ d }))
+							.catch(/** @param {any} e */ (e) => ({ e }))
+							.then(
+								/**
+								 * @param {{d: any; e: any}} result
+								 */
+								async ({ d, e }) => {
+									let data;
+									let error;
+									try {
+										if (e) {
+											error = devalue.stringify(e, revivers);
+										} else {
+											data = devalue.stringify(d, revivers);
+										}
+									} catch (e) {
+										error = await handle_error_and_jsonify(
+											event,
+											options,
+											new Error(clarify_devalue_error(event, /** @type {any} */ (e)))
+										);
+									}
+
+									if (
+										__SVELTEKIT_DEV__ &&
+										stringify_uses(/** @type {import('types').ServerDataNode} */ (node)) !==
+											uses_str
+									) {
+										console.warn(
+											'Accessed dependencies after load function returned. These usages will not be tracked.'
```

Comment:
This warning needs a better text - best I could come up with on a friday evening